build.sbt

```diff
@@ -62,13 +62,18 @@ lazy val core = projectMatrix
     libraryDependencies ++= {
       if (isScala3.value) {
         List(
-          scalameta
+          scalameta,
+          // CrossVersion.for3Use2_13 would only lookup a binary version artifact, but this is published with full version
+          semanticdbScalacCore.cross(CrossVersion.constant(scala213))
```

Comment:
Following https://github.com/scalameta/scalameta/pull/3747, this was added to continue building against `internal.{symtab,metap}` (now directly via semanticdb-scalac-core) & protobuf stubs in `internal.semanticdb` (now transitively via semanticb-shared). Unfortunately, this brings way too much stuff: `scala.meta.interactive, other unused (internal) classes, and the Scala2 compiler itself.

If this turns out to be a problem, we could either (1) revisit this upstream, or (2) rely on semanticdb-shared and shade only packages we need from  semanticdb-scalac-core.

Reply:
The biggest problem here I guess is that we are using internal packages, we should move them out of internal if they are intended to be binary compat or as you suggest shade them in scalafix :thinking:

Reply:
Indeed, I will go for (2). For the record: I'll wait for the next scalameta release I guess (since I am effectively blocked by https://github.com/scalameta/scalameta/pull/3786#issuecomment-2267650358), where [semanticdb-shared will be cross-built with binary version only](https://github.com/scalameta/scalameta/pull/3887).

Reply:
it was not that easy in the end :( https://github.com/scalacenter/scalafix/pull/2048